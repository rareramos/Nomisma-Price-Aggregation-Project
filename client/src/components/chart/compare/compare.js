import React from 'react';
import { func, bool } from 'prop-types';
import { Switch } from '@nomisma/nomisma-ui/charts/switch';

export const ComapreComp = ({ isInComapreMode, onClick }) => (
  <Switch
    checked={isInComapreMode}
    onChange={() => onClick(!isInComapreMode)}
  />
);

ComapreComp.propTypes = {
  isInComapreMode: bool.isRequired,
  onClick: func.isRequired,
};
