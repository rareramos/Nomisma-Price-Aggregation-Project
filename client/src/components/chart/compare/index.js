export { Compare } from './compare-container';
