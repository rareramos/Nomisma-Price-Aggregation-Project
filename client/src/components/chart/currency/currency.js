import React from 'react';
import {
  func, arrayOf, shape, string,
} from 'prop-types';
import { ChartTabs } from '../../common/chart-tabs';
import { currencies as currenciesMock } from '../../../utils/chart';

export const CurrenciesComp = ({ currencies, selectedSource, onClick }) => (
  <ChartTabs
    selected={selectedSource}
    setActiveNavItem={value => onClick(value)}
    navItems={currencies}
  />
);

CurrenciesComp.defaultProps = {
  currencies: currenciesMock,
};

CurrenciesComp.propTypes = {
  selectedSource: string.isRequired,
  currencies: arrayOf(shape({ title: string, type: string })),
  onClick: func.isRequired,
};
