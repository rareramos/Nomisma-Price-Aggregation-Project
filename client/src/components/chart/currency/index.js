export { Currencies } from './currency-container';
