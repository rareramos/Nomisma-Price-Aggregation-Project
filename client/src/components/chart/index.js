export { Chart } from './chart-container';
