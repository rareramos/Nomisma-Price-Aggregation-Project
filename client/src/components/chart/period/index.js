export { Period } from './period-container';
