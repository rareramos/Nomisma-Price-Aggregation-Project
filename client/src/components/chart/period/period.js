import React from 'react';
import {
  func, arrayOf, shape, string,
} from 'prop-types';
import { ChartTabs } from '../../common/chart-tabs';
import { periods as periodMock } from '../../../utils/chart';

export const Period = ({ periods, selectedSource, onClick }) => (
  <ChartTabs
    selected={selectedSource}
    setActiveNavItem={value => onClick(value)}
    navItems={periods}
  />
);

Period.defaultProps = {
  periods: periodMock,
};
Period.propTypes = {
  selectedSource: string.isRequired,
  periods: arrayOf(shape({ title: string, type: string })),
  onClick: func.isRequired,
};
