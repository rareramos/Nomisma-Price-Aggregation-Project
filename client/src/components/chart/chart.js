import React, { Component } from 'react';
import {
  arrayOf, func, string, bool,
  shape, instanceOf, number,
} from 'prop-types';
import { StrikeChart } from '@nomisma/nomisma-ui/charts/strike-chart';
import { Grid } from '@nomisma/nomisma-ui/grid';
import { Period } from './period';
import { Source } from './source';
import { Currencies } from './currency';
import { Compare } from './compare';

export class ChartView extends Component {
  static capitalize(stringValue) {
    return stringValue.charAt(0).toUpperCase() + stringValue.slice(1);
  }

  componentWillMount() {
    const {
      period, fetchChartData,
    } = this.props;

    fetchChartData('all', period);
  }

  render() {
    const {
      compare,
      loading,
      currency,
      protocol,
      chartData,
      chartDataSecondary,
    } = this.props;
    return (
      <div>
        <Grid
          column="160px 328px 196px"
          columnGap="256px"
        >
          <Currencies />
          <Source />
          <Period />
        </Grid>
        <StrikeChart
          id="chartdiv"
          data={chartData}
          width="100%"
          height="400px"
          currency={currency.toUpperCase()}
          hideFunctions
          loading={loading}
          dataName={compare ? 'Compound' : ChartView.capitalize(protocol)}
          secondaryDataSet={chartDataSecondary}
          secondaryDataName={chartDataSecondary.length > 0 && 'Dharma'}
        />
        <Grid
          column="auto 100px"
        >
          <Grid
            column="90px 50px"
          >
            <div>Comapare:</div>
            <Compare />
          </Grid>
        </Grid>
      </div>
    );
  }
}

ChartView.defaultProps = {
  chartData: [],
  chartDataSecondary: [],
  protocol: 'all',
  period: '24h',
  loading: true,
  currency: 'usd',
  compare: false,
};

ChartView.propTypes = {
  fetchChartData: func.isRequired,
  chartData: arrayOf(shape({ date: instanceOf(Date), value: number })),
  chartDataSecondary: arrayOf(shape({ date: instanceOf(Date), value: number })),
  protocol: string,
  period: string,
  loading: bool,
  currency: string,
  compare: bool,
};
