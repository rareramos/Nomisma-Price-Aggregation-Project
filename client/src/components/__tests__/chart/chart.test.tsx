import * as React from 'react';
import { shallow } from 'enzyme';

import { ComapreComp } from 'components/chart/compare/compare';
import { CurrenciesComp } from 'components/chart/currency/currency';
import { Period } from 'components/chart/period/period';
import { SourceComp } from 'components/chart/source/source';
import { ChartView } from 'components/chart/chart';
import { currencies, periods, sources } from 'utils';
import {
  IChartCompareProps,
  IChartCurrenciesProps,
  IChartPeriodProps,
  IChartSourceProps,
  IChartViewProps,
} from 'types/components/chart';

const CompareChartProps : IChartCompareProps = {
  isInComapreMode: false,
  onClick: jest.fn,
};

const CurrencyChartProps : IChartCurrenciesProps = {
  currencies,
  onClick: jest.fn,
  selectedSource: 'usd',
};

const SourceChartProps : IChartSourceProps = {
  onClick: jest.fn,
  selectedSource: 'usd',
  sources,
};

const PeriodChartProps : IChartPeriodProps = {
  onClick: jest.fn,
  periods,
  selectedSource: 'usd',
};

const ChartViewProps : IChartViewProps = {
  chartData: [],
  chartDataSecondary: [],
  compare: false,
  currency: 'usd',
  fetchChartData: jest.fn,
  period: '86400',
  protocol: 'all',
};

describe('chart tests', () => {
  it('matches compare snapshot', () => {
    const wrapper = shallow(<ComapreComp {...CompareChartProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches currency snapshot', () => {
    const wrapper = shallow(<CurrenciesComp {...CurrencyChartProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches source snapshot', () => {
    const wrapper = shallow(<SourceComp {...SourceChartProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches period snapshot', () => {
    const wrapper = shallow(<Period {...PeriodChartProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches chart view snapshot', () => {
    const wrapper = shallow(<ChartView {...ChartViewProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
