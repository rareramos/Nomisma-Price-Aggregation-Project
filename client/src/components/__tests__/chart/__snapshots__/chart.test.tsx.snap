// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chart tests matches chart view snapshot 1`] = `
<div>
  <styled.div
    column="160px 328px 196px"
    columnGap="256px"
  >
    <Connect(CurrenciesComp) />
    <Connect(SourceComp) />
    <Connect(Period) />
  </styled.div>
  <StrikeChart
    currency="USD"
    data={Array []}
    dataName="All"
    height="400px"
    hideFunctions={true}
    id="chartdiv"
    loading={true}
    secondaryDataName={false}
    secondaryDataSet={Array []}
    width="100%"
  />
  <styled.div
    column="auto 100px"
  >
    <styled.div
      column="90px 50px"
    >
      <div>
        Comapare:
      </div>
      <Connect(ComapreComp) />
    </styled.div>
  </styled.div>
</div>
`;

exports[`chart tests matches compare snapshot 1`] = `
<Switch
  checked={false}
  onChange={[Function]}
/>
`;

exports[`chart tests matches currency snapshot 1`] = `
<ChartTabs
  navItems={
    Array [
      Object {
        "title": "USD",
        "type": "usd",
      },
      Object {
        "title": "ETH",
        "type": "eth",
      },
      Object {
        "title": "BTC",
        "type": "btc",
      },
    ]
  }
  selected="usd"
  setActiveNavItem={[Function]}
/>
`;

exports[`chart tests matches period snapshot 1`] = `
<ChartTabs
  navItems={
    Array [
      Object {
        "title": "24h",
        "type": "86400",
      },
      Object {
        "title": "7d",
        "type": "604800",
      },
      Object {
        "title": "30d",
        "type": "2592000",
      },
      Object {
        "title": "6m",
        "type": "15552000",
      },
    ]
  }
  selected="usd"
  setActiveNavItem={[Function]}
/>
`;

exports[`chart tests matches source snapshot 1`] = `
<ChartTabs
  navItems={
    Array [
      Object {
        "title": "All",
        "type": "all",
      },
      Object {
        "title": "Compound",
        "type": "compound",
      },
      Object {
        "title": "Dharma",
        "type": "dharma",
      },
      Object {
        "title": "Maker DAO",
        "type": "maker",
      },
    ]
  }
  selected="usd"
  setActiveNavItem={[Function]}
/>
`;
